lib/ipmi_hpeoem.c

```diff
@@ -0,0 +1,428 @@
+            return -1;
+        }
+        if (str2uchar(argv[current_arg], &channel) != 0)
+        {
+            lprintf(LOG_ERR, "Invalid channel: %s", argv[current_arg]);
+            return -1;
+        }
+        ipmi_hpeoem_lan_get_nic_selection(intf, channel);
+    }
+    else
+    {
+        ipmi_hpeoem_lan_usage();
+        return -1;
+    }
+    return rc;
+}
+
+int ipmi_hpeoem_main(struct ipmi_intf *intf, int argc, char **argv)
+{
+    int rc = 0;
+    current_arg = 0;
+    if (!argc || !strcmp(argv[0], "help"))
+    {
+        usage();
+        return 0;
+    }
+    if (!strcmp(argv[current_arg], "lan"))
+    {
+        /* lan address */
+        rc = ipmi_hpeoem_lan_main(intf, argc, argv);
+    }
+    else
+    {
+        usage();
+        return -1;
+    }
+    return rc;
+}
+
+static void
+usage(void)
+{
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "usage: hpeoem <command> [option...]");
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "commands:");
+    lprintf(LOG_NOTICE,
+            "    lan");
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "For help on individual commands type:");
+    lprintf(LOG_NOTICE,
+            "hpeoem <command> help");
+}
+
+static void
+ipmi_hpeoem_lan_usage(void)
+{
+    lprintf(LOG_NOTICE,
+            "");
+    lprintf(LOG_NOTICE,
+            "   lan set <channel> <mode>");
+    lprintf(LOG_NOTICE,
+            "      sets the NIC Selection Mode :");
+    lprintf(LOG_NOTICE,
+            "            dedicated, shared with lom1, shared with lom2, shared with alom1,");
+    lprintf(LOG_NOTICE,
+            "            shared with alom2");
+    lprintf(LOG_NOTICE,
+            "   lan get <channel>");
+    lprintf(LOG_NOTICE,
+            "      returns the current active NIC.");
+    lprintf(LOG_NOTICE,
+            "");
+}
+
+static int
+ipmi_hpeoem_lan_get_nic_selection(struct ipmi_intf *intf, int chan)
+{
+    struct ipmi_rs *rsp;
+    struct ipmi_rq req;
+    uint8_t input_length = 4;
+    uint8_t msg_data[4];
+    uint8_t nic_selection = -1;
+    uint8_t shared_nic_selection = -1;
+    uint8_t shared_nic_port = -1;
+    char *nic_selection_str = "Unknown";
+    char *shared_nic_selection_str = "Unknown";
+    char *shared_nic_port_str = "Unknown";
+    uint8_t reset_needed = 0;
+
+    msg_data[0] = chan;
+    msg_data[1] = IPMI_ILO_NETWORK_CONFIG_NIC_SELECTION;
+    msg_data[2] = 0;
+    msg_data[3] = 0;
+
+    req.msg.netfn = IPMI_NETFN_TRANSPORT;
+    req.msg.cmd = IPMI_LAN_GET_CONFIG;
+    req.msg.lun = 0;
+    req.msg.data = msg_data;
+    req.msg.data_len = input_length;
+
+    rsp = intf->sendrecv(intf, &req);
+    if (!rsp)
+    {
+        lprintf(LOG_ERR, "Error in getting nic selection");
+        return -1;
+    }
+    else if (rsp->ccode)
+    {
+        lprintf(LOG_ERR, "Error in getting nic selection (%s)",
+                val2str(rsp->ccode, completion_code_vals));
```

Comment:
`IPMI_LAN_GET_CONFIG` and `IPMI_LAN_SET_CONFIG` can return command-specific codes that aren't covered by `completion_code_vals`. Check `get_lan_cc_vals` in `lib/ipmi_lanp.c`.

I think that instead of reimplementing LAN parameter setting and getting here, you could just reuse the existing `get_lan_param()` and `set_lan_param_nowait()` functions (you'll need to make those public).